etna/analysis/eda_utils.py

```diff
@@ -189,30 +194,72 @@ def sample_acf_plot(
         number of random segments to plot
     lags:
         number of timeseries shifts for cross-correlation
+    partial:
+        plot autocorrelation or partial autocorrelation
+    columns_num:
+        number of columns in subplots
     segments:
         segments to plot
     figsize:
         size of the figure per subplot with one segment in inches
     """
+    df_pd = ts.to_pandas(flatten=True)
     if segments is None:
-        segments = sorted(ts.segments)
+        exist_segments = df_pd.segment.unique()
```

Comment:
Why can't we write this if like:
```
if segments is None:
    segments = sorted(ts.segments)
    selected_segments = np.random.choice(...)
    segments = list(selected_segments)
```

In this case we don't need `df_pd` variable.